docs/rst/recording/user_interface/configuration.rst

```diff
@@ -0,0 +1,322 @@
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This is the list of topics that |ddsrecorder| will forward, i.e. the data published under the topics matching the
+expressions in the ``allowlist`` will be relayed by |ddsrecorder|.
+
+.. note::
+
+    If no ``allowlist`` is provided, data will be forwarded for all topics (unless filtered out in ``blocklist``).
+
+
+Block topic list (``blocklist``)
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+This is the list of topics that the |ddsrecorder| will block, that is, all data published under the topics matching the
+filters specified in the ``blocklist`` will be discarded by the |ddsrecorder| and therefore will not be relayed.
+
+This list takes precedence over the ``allowlist``.
+If a topic matches an expression both in the ``allowlist`` and in the ``blocklist``, the ``blocklist`` takes precedence,
+causing the data under this topic to be discarded.
+
+
+Examples of usage
+^^^^^^^^^^^^^^^^^
+
+The following is an example of how to use the ``allowlist``, ``blocklist`` and ``builtin-topics`` configurations to
+setup the |ddsrecorder| filtering rules.
+
+Dynamic topic discovery example
+"""""""""""""""""""""""""""""""
+
+This example shows how the |ddsrecorder| is initially configured to forward the ``rt/chatter`` topic (default ROS 2
+topic for ``talker`` and ``listener``) with type name ``std_msgs::msg::dds_::String_``, while the rest of the
+topics in the DDS network are expected to be dynamically discovered.
+Additionally, two rules are specified in the ``blocklist`` in order to filter out messages of no interest to the user
+(in this case ROS2 services related topics).
+
+.. code-block:: yaml
+
+    builtin-topics:
+      - name: rt/chatter
+        type: std_msgs::msg::dds_::String_
+
+    blocklist:
+      - name: "rq/*"
+      - name: "rr/*"
+
+
+Allowlist and blocklist collision
+"""""""""""""""""""""""""""""""""
+
+In the following example, the ``HelloWorldTopic`` topic is both in the ``allowlist`` and (implicitly) in the
+``blocklist``, so according to the ``blocklist`` preference rule this topic is blocked.
+Moreover, only the topics present in the allowlist are relayed, regardless of whether more topics are dynamically
+discovered in the DDS network.
+In this case the forwarded topics are ``AllowedTopic1`` with data type ``Allowed``
+and ``AllowedTopic2`` regardless of its data type.
+
+.. code-block:: yaml
+
+    allowlist:
+      - name: AllowedTopic1
+        type: Allowed
+      - name: AllowedTopic2
+        type: "*"
+      - name: HelloWorldTopic
+        type: HelloWorld
+
+    blocklist:
+      - name: "*"
+        type: HelloWorld
+
+.. _user_interface_configuration_domain_id:
+
+Domain Id
+---------
+
+Tag ``domain`` configures the :term:`Domain Id`.
+
+.. code-block:: yaml
+
+    domain: 101
+
+
+.. _user_interface_configuration_general_example:
+
+General Example
+---------------
+
+A complete example of all the configurations described on this page can be found below.
+
+.. code-block:: yaml
```

Comment:
1. Remove the `builtin-topics` from this configuration file. 
2. Add an `allowlist`
3. Add file configuration parameters (you have them in the previous comment)